httpx/http_service.go

```diff
@@ -0,0 +1,165 @@
+package httpx
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/remind101/pkg/retry"
+
+	"golang.org/x/net/context"
+)
+
+type RoundTripper interface {
+	RoundTrip(*http.Request, context.Context) (*http.Response, error)
+}
+
+type Client struct {
+	innerClient		*http.Client
+	retryTransport 	*RetryTransport
+	baseURL 		*url.URL
+}
+
+type RetryTransport struct {
+	Retrier     *retry.Retrier
```

Comment:
Oh, and it looks like you're not running gofmt on save. Might be good to setup something like goimports in your editor to run on save: https://github.com/bradfitz/goimports